### ISSUE TITLE
Forbid `lambda` to be in comprehension value part

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1223]
# Rule request

## Thesis

This code: 

```python
>>> x = [lambda y: a + y for a in [1, 2, 3]]
>>> x
[<function <listcomp>.<lambda> at 0x10c2e6b80>, <function <listcomp>.<lambda> at 0x10c3551f0>, <function <listcomp>.<lambda> at 0x10c355280>]
>>> x[0](1)
4
```

Should raise `WPS426`. But it is not raising it.

## Reasoning

It still gives the wrong result.
